Purchases/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -69,20 +75,22 @@ class StoreKit2TransactionListener {
 @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
 private extension StoreKit2TransactionListener {
 
-    func handle(transactionResult: VerificationResult<StoreKit.Transaction>) async {
+    /// - Returns: whether the transaction was verified
+    func handle(transactionResult: VerificationResult<StoreKit.Transaction>) async throws -> Bool {
         switch transactionResult {
-        case .unverified(let unverifiedTransaction, let verificationError):
-            // todo: update once StoreKit fixes the issue with verifying sandbox purchases.
-            #if DEBUG
-            Logger.debug("The transaction has failed verification, but Sandbox purchases dont' support" +
-                         "verification as of Beta 8. Details: \(verificationError.localizedDescription)")
-            await finish(transaction: unverifiedTransaction)
-            #else
-            Logger.error("StoreKit has parsed the JWS but failed verification. The content will not be" +
-                         "made available to the user. Details: \(verificationError.localizedDescription)")
-            #endif
+        case let .unverified(unverifiedTransaction, verificationError):
+            throw ErrorUtils.storeProblemError(
+                withMessage: Strings.purchase.transaction_unverified(
+                    productID: unverifiedTransaction.productID,
+                    errorMessage: verificationError.localizedDescription
+                ).description,
+                error: verificationError
+            )
+
         case .verified(let verifiedTransaction):
             await finish(transaction: verifiedTransaction)
+
+            return true
```

Comment:
I'm a bit confused by this method - we have both a bool return and an error, but if something goes wrong we throw an error and don't return false, right? perhaps we should keep only the error?

Reply:
The result indicates a cancellation.

Reply:
but... there's no `return false` here, right? also this method says 
```swift
/// - Returns: whether the transaction was verified
```

aren't cancellations handled in `func handle(purchaseResult: StoreKit.Product.PurchaseResult)`?

Reply:
Sorry finally looking at this.
You're totally right 👍 I refactored this and didn't realize the return is redundant now.